src/daft-core/src/series/mod.rs

```diff
@@ -38,6 +46,58 @@ impl PartialEq for Series {
 }
 
 impl Series {
+    /// Build a hashset of the [`IndexHash`]s of each element in this [`Series`].
+    ///
+    /// The returned hashset can be used to probe for the existence of a given element in this [`Series`].
+    /// Its length can also be used to determine the *exact* number of unique elements in this [`Series`].
+    ///
+    /// # Note
+    /// 1. This function returns a `HashMap<X, ()>` rather than a `HashSet<X>`. These two types are functionally equivalent.
+    ///
+    /// 2. `NULL`s are *not* inserted into the returned hashset. They won't be counted towards the final number of unique elements.
+    pub fn build_probe_table(&self) -> DaftResult<HashMap<IndexHash, (), IdentityBuildHasher>> {
```

Comment:
Maybe we should rename to something else since it does something that is slightly different from the other parts of the code that build probe tables - it excludes null rows entirely. 

Alternatively, we could maybe provide an argument for whether to exclude nulls. If we do that, we could even reuse this in the probe set construction that's done in the swordfish hash semi/anti joins

Reply:
I think I like the second route. I'll add an additional "flag" parameter to determine if `NULL`s should be included or not.

Reply:
Actually, I opted for the first route. Let's get this PR into main ASAP so that it unblocks other stuff. In the future, I'll we can extend this for Swordfish to use.

Reply:
Updated.